pkg/ipmasq/ipmasq_test.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !privileged_tests
+
+package ipmasq
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	"gopkg.in/check.v1"
+
+	"github.com/cilium/cilium/pkg/controller"
+)
+
+func Test(t *testing.T) {
+	check.TestingT(t)
+}
+
+type ipMasqMapMock struct {
+	cidrs map[string]net.IPNet
+}
+
+func (m *ipMasqMapMock) Update(cidr net.IPNet) error {
+	cidrStr := cidr.String()
+	if _, ok := m.cidrs[cidrStr]; ok {
+		return fmt.Errorf("CIDR already exists: %s", cidrStr)
+	}
+	m.cidrs[cidrStr] = cidr
+	return nil
+}
+
+func (m *ipMasqMapMock) Delete(cidr net.IPNet) error {
+	cidrStr := cidr.String()
+	if _, ok := m.cidrs[cidrStr]; !ok {
+		return fmt.Errorf("CIDR not found: %s", cidrStr)
+	}
+	delete(m.cidrs, cidrStr)
+	return nil
+}
+
+func (m *ipMasqMapMock) Dump() ([]net.IPNet, error) {
+	cidrs := make([]net.IPNet, 0, len(m.cidrs))
+	for _, cidr := range m.cidrs {
+		cidrs = append(cidrs, cidr)
+	}
+	return cidrs, nil
+}
+
+type IPMasqTestSuite struct {
+	ipMasqMap  *ipMasqMapMock
+	manager    *controller.Manager
+	configFile *os.File
+}
+
+var _ = check.Suite(&IPMasqTestSuite{})
+
+func (i *IPMasqTestSuite) SetUpTest(c *check.C) {
+	i.ipMasqMap = &ipMasqMapMock{cidrs: map[string]net.IPNet{}}
+	i.manager = controller.NewManager()
+
+	configFile, err := ioutil.TempFile("/tmp", "ipmasq-test")
```

Comment:
There's no need to specify the tmp directory according to the ioutil API documentation:

```suggestion
	configFile, err := ioutil.TempFile("", "ipmasq-test")
```